bcda/cclf/cclf.go

```diff
@@ -82,42 +79,24 @@ func (importer CclfImporter) importCCLF0(ctx context.Context, fileMetadata *cclf
 	close := metrics.NewChild(ctx, "importCCLF0")
 	defer close()
 
-	var validator map[string]cclfFileValidator
-	var rawFile *zip.File
-
-	for _, f := range r.File {
-		// iterate in this zipped folder until we find our cclf0 file
-		if f.Name == fileMetadata.name {
-			rawFile = f
-			importer.Logger.Infof("Reading file %s from archive %s", fileMetadata.name, fileMetadata.filePath)
-		}
-	}
-
-	if rawFile == nil {
-		err = errors.Wrapf(err, constants.FileNotFound, fileMetadata.name, fileMetadata.filePath)
-		importer.Logger.Error(err)
-		return nil, err
-	}
-
-	rc, err := rawFile.Open()
+	rc, err := zipMetadata.cclf0File.Open()
 	if err != nil {
-		err = errors.Wrapf(err, "could not read file %s in CCLF0 archive %s", fileMetadata.name, fileMetadata.filePath)
+		err = errors.Wrapf(err, "could not read file %s in CCLF0 archive %s", fileMetadata.name, zipMetadata.filePath)
 		importer.Logger.Error(err)
 		return nil, err
 	}
 	defer rc.Close()
 	sc := bufio.NewScanner(rc)
+
+	var validator *cclfFileValidator
+
 	for sc.Scan() {
 		b := sc.Bytes()
 		if len(bytes.TrimSpace(b)) > 0 {
 			filetype := string(bytes.TrimSpace(b[fileNumStart:fileNumEnd]))
 
 			if filetype == "CCLF8" {
-				if validator == nil {
-					validator = make(map[string]cclfFileValidator)
-				}
-
-				if _, ok := validator[filetype]; ok {
```

Comment:
Replaced the map of validators with just a single validator instance, since we only care about the CCLF8 validation.